tests/zfs-tests/include/libtest.shlib

```diff
@@ -2692,6 +2785,80 @@ function get_device_dir #device
 }
 
 #
+# Extract real disk variables from given disk
+#
+# Returns HBA CHAN TARG LUN
+function get_disk_variables #disk
+{
+	typeset disk=$1
+	typeset hba=""
+	typeset chan=""
+	typeset targ=""
+	typeset lun=""
+	typeset dm_name=""
+
+	[[ -z $disk ]] \
+	    && log_fail "Argument missing or invalid"
+	if is_linux; then
+		if ( is_mpath_device $disk ); then
+			dm_name="$($READLINK $DEV_DSKDIR/$disk \
+			    | $NAWK -F / '{print $2}')"
+			slave="$($LS /sys/block/$dm_name/slaves \
+			    | $NAWK '{print $1}')"
+			hba="$($LS /sys/block/$dm_name/slaves/$slave/device/scsi_device/ \
+			    | $NAWK -F : '{print $1}')"
+			chan="$($LS /sys/block/$dm_name/slaves/$slave/device/scsi_device/ \
+			    | $NAWK -F : '{print $2}')"
+			targ="$($LS /sys/block/$dm_name/slaves/$slave/device/scsi_device/ \
+			    | $NAWK -F : '{print $3}')"
+			lun="$($LS /sys/block/$dm_name/slaves/$slave/device/scsi_device/ \
+			    | $NAWK -F : '{print $4}')"
+		elif ( is_real_device $disk ); then
+			hba="$($LS /sys/block/$disk/device/scsi_device/ \
+			    | $NAWK -F : '{print $1}')"
+			chan="$($LS /sys/block/$disk/device/scsi_device/ \
+			    | $NAWK -F : '{print $2}')"
+			targ="$($LS /sys/block/$disk/device/scsi_device/ \
+			    | $NAWK -F : '{print $3}')"
+			lun="$($LS /sys/block/$disk/device/scsi_device/ \
+			    | $NAWK -F : '{print $4}')"
+		else
+			log_fail "get_disk_variables() failure"
+		fi
+		$ECHO "$hba $chan $targ $lun"
+	else
+		log_fail
+	fi
+}
+
+#
+# Get persistent name for given disk
+#
+function get_persistent_disk_name #device
+{
+	typeset device=$1
+	typeset dev_id
+
+	if is_linux; then
+		if ( is_real_device $device ); then
+			dev_id="$($UDEVADM info $DEV_DSKDIR/$device | $EGREP \
+			    disk/by-id | $NAWK '{print $2; exit}' \
+			    | $NAWK -F / '{print $3}')"
```

Comment:
I suspect this won't work on CentOS6 due to the older version of udevadm but we'll have to try it and see.